### ISSUE TITLE
PHP 8 compatibility

### ISSUE THREAD

**Issue by kevin-appelt** [Tuurlijk/typo3scan#43]
It would be great if the scanner also supports PHP 8.
I haven't dived deeper yet but the "biggest" problem is that "match" is a reserved keyword in PHP 8 which leads to the following error:
```
Parse error: syntax error, unexpected token "::", expecting "(" in phar:///var/www/html/vendor/michielroos/typo3scan/typo3scan/vendor/typo3/cms-scanner/src/Scanner.php on line 25
PHP Parse error:  syntax error, unexpected token "::", expecting "(" in phar:///var/www/html/vendor/michielroos/typo3scan/typo3scan/vendor/typo3/cms-scanner/src/Scanner.php on line 25
```

**Comment by Tuurlijk** [comment]
Sounds like a minor change. Could be renamed to $theMatch or something.

**Comment by AKaravas** [comment]
i get the same problem running the following:

`php typo3scan.phar scan --target 11 extensions/my_extension `

> PHP Parse error:  syntax error, unexpected token "::", expecting "(" in phar://D:/Programming/MyPath/typo3scan.phar/vendor/typo3/cms-scanner/src/Scanner.php on line 25

Is there any plan to fix this?

**Comment by mbrodala** [comment]
Suggested fix for now: use `^7.0` for PHP. Then extend this to `^7.0 || ^8.0` once support for PHP 8 is added.
